File: 94_Binary_Tree_Inorder_Traversal/inorderTraversal.py
class Solution(object):
    def inorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        stack = []
        def dfs(root):
            if not root: return
            if root.left:
                dfs(root.left)
            stack.append(root.val)
            if root.right:
                dfs(root.right)
        dfs(root)
        return stack

File: 94_Binary_Tree_Inorder_Traversal/inorderTraversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> res;
        stack<TreeNode*> st;
        
        while(root) {
            st.push(root);
            root = root->left;
        }
        
        while(!st.empty()) {
            TreeNode* curr = st.top();
            st.pop();
            res.push_back(curr->val);
            TreeNode* node = curr->right;
            while(node) {
                st.push(node);
                node = node->left;
            }
        }
        
        return res;
    }
};